cl_sii/rcv/data_models.py

```diff
@@ -214,79 +227,109 @@ def as_dte_data_l2(self) -> cl_sii.dte.data_models.DteDataL2:
-            raise TypeError("Inappropriate type of 'receptor_razon_social'.")
-        cl_sii.dte.data_models.validate_contribuyente_razon_social(self.receptor_razon_social)
-
-        if self.fecha_acuse_dt is not None:
-            if not isinstance(self.fecha_acuse_dt, datetime):
-                raise TypeError("Inappropriate type of 'fecha_acuse_dt'.")
-            tz_utils.validate_dt_tz(self.fecha_acuse_dt, SII_OFFICIAL_TZ)
-
-        if self.fecha_reclamo_dt is not None:
-            if not isinstance(self.fecha_reclamo_dt, datetime):
-                raise TypeError("Inappropriate type of 'fecha_reclamo_dt'.")
-            tz_utils.validate_dt_tz(self.fecha_reclamo_dt, SII_OFFICIAL_TZ)
+    fecha_reclamo_dt: Optional[datetime] = None
 
+    ###########################################################################
+    # Validators
+    ###########################################################################
 
-@dataclasses.dataclass(frozen=True)
+    @pydantic.validator('receptor_razon_social')
+    def validate_contribuyente_razon_social(cls, v: object) -> object:
+        if isinstance(v, str):
+            cl_sii.dte.data_models.validate_contribuyente_razon_social(v)
+        return v
+
+    @pydantic.validator('fecha_acuse_dt', 'fecha_reclamo_dt')
+    def validate_datetime_tz(cls, v: object) -> object:
+        if isinstance(v, datetime):
+            tz_utils.validate_dt_tz(v, cls.DATETIME_FIELDS_TZ)
+        return v
+
+
+@pydantic.dataclasses.dataclass(
+    frozen=True,
+    config=type('Config', (), dict(
+        arbitrary_types_allowed=True,
+    ))
+)
 class RcRegistroDetalleEntry(RcvDetalleEntry):
 
     """
     Entry of the "detalle" of an RC ("Registro de Compras") / "registro".
     """
 
+    ###########################################################################
+    # constants
+    ###########################################################################
+
+    DATETIME_FIELDS_TZ = SII_OFFICIAL_TZ
+
     RCV_KIND = RcvKind.COMPRAS
     RC_ESTADO_CONTABLE = RcEstadoContable.REGISTRO
 
-    emisor_razon_social: str = dc_field()
+    emisor_razon_social: str
     """
     "Razón social" (legal name) of the "emisor" of the "documento".
     """
 
     # TODO: docstring
     # note: must be timezone-aware.
-    fecha_acuse_dt: Optional[datetime] = dc_field()
-
-    def __post_init__(self) -> None:
-        super().__post_init__()
-
-        if not isinstance(self.emisor_razon_social, str):
-            raise TypeError("Inappropriate type of 'emisor_razon_social'.")
-        cl_sii.dte.data_models.validate_contribuyente_razon_social(self.emisor_razon_social)
-
-        if self.fecha_acuse_dt is not None:
-            if not isinstance(self.fecha_acuse_dt, datetime):
-                raise TypeError("Inappropriate type of 'fecha_acuse_dt'.")
-            tz_utils.validate_dt_tz(self.fecha_acuse_dt, SII_OFFICIAL_TZ)
+    fecha_acuse_dt: Optional[datetime] = None
```

Comment:
See review comment for field `RvDetalleEntry.fecha_acuse_dt`.

```suggestion
    fecha_acuse_dt: Optional[datetime]
```